src/connection-pool-party.js

```diff
@@ -461,7 +484,9 @@ export default class ConnectionPoolParty extends EventEmitter {
       return Promise.resolve(false);
     }
     // get any updated dsn info from the provider
-    this._healingPromise = this._healingPromise || this.dsnProvider()
+    this._healingPromise = this._healingPromise || Promise.resolve()
+      .then(() => debug('healing started, retrieving new dsns from provider'))
```

Comment:
wouldn't these only apply to the `Promise.resolve()`? the `||` needs some parens it looks like.

Reply:
that's the intent here. it's a mechanism to ensure the healing code is only run once but still allow multiple calls to wait for the resolution

Reply:
just confusing to read, putting () around the one it is intended to apply to would at least show it is intentional

Reply:
ah i see, yea that's reasonable, will add